server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -170,6 +172,11 @@ public int countByQuery(DbSession session, String organizationUuid, ComponentQue
     return executeLargeInputs(keys, mapper(session)::selectByKeys);
   }
 
+  public List<ComponentDto> selectByKeysAndBranch(DbSession session, Collection<String> keys, String branch) {
+    List<String> dbKeys = keys.stream().map(k -> generateBranchKey(k, branch)).collect(MoreCollectors.toList());
+    return executeLargeInputs(dbKeys, mapper(session)::selectByDbKeys);
+  }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1612)